src/scatter_buildblock/ScatterEstimationByBin.cxx

```diff
@@ -11,447 +11,1274 @@
+
+            this->input_projdata_2d_sptr.reset(new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                                                     this->proj_data_info_2d_sptr,
+                                                                     out_filename,
+                                                                     std::ios::in | std::ios::out | std::ios::trunc));
+        }
+        else
+            this->input_projdata_2d_sptr.reset(new ProjDataInMemory(this->input_projdata_sptr->get_exam_info_sptr(),
+                                                                    this->proj_data_info_2d_sptr));
+
+        SSRB(*this->input_projdata_2d_sptr,
+             *input_projdata_sptr,false);
+    }
+    else
+    {
+        warning(boost::format("The input data %1% are not 3D. Abord.") %this->input_projdata_filename );
+        return Succeeded::no;
+    }
+
+    //
+    // Load the initial activity image if !recomput_initial_activity_image, else
+    // Create and initialise the activity image (and run reconstruction::set_up())
+    //
+    if (!this->recompute_initial_activity_image && this->initial_activity_image_filename.size() > 0 )
+    {
+        info("Loading initial activity image ...");
+        this->activity_image_sptr =
+                read_from_file<DiscretisedDensity<3,float> >(this->initial_activity_image_filename);
+    }
+    else
+    {
+        info("Initialising empty activity image ... ");
+        this->activity_image_sptr.reset(new VoxelsOnCartesianGrid<float> ( *this->input_projdata_2d_sptr->get_proj_data_info_sptr() ) );
+        this->activity_image_sptr->fill(1.F);
+    }
+
+    //
+    // Beside the attenuation correction we need it for the mask and the
+    // ScatterSimulation
+    //
+    {
+        if (is_null_ptr(this->atten_image_sptr))
+        {
+            warning("Attenuation image has not been loaded properly. Abord.");
+            return Succeeded::no;
+        }
+
+        info(boost::format("Attenuation image data are supposed to be in units cm^-1\n"
+                           "\tReference: water has mu .096 cm^-1\n"
+                           "\tMax in attenuation image: %g\n") %
+             this->atten_image_sptr->find_max());
+
+        int min_z = this->atten_image_sptr->get_min_index();
+        int min_y = this->atten_image_sptr.get()[0][min_z].get_min_index();
+        int len_y = this->atten_image_sptr.get()[0][min_z].get_length();
+        int len_x = this->atten_image_sptr.get()[0][min_z][min_y].get_length();
+
+        if (len_y != len_x)
+            error(boost::format("The voxels in the x (%1%) and y (%2%)"
+                                " are different. Cannot zoom...  ") %len_x % len_y );
+    }
+
+    //
+    // Zoom the activity image
+    // TODO: add option to opt-out
+    //
+    {
+        int size_xy = 0;
+        int size_z = 0;
+        // zoom image
+        float zoom_z = 1.0f;
```

Comment:
we have to get rid of the hard-wired values. I thought we had a scheme for that, maybe you still have to commit this. Otherwise just make them 2 parsing parameters.